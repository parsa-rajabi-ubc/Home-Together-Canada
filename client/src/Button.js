import React from 'react';

export function Button(props){
    return(
        <label>
            {props.label}
            <input type="button" value={props.value} />
        </label>
    );
}