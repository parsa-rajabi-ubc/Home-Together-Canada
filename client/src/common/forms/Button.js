import React from 'react';
import PropTypes from "prop-types";

//returns generic Button field for a form with a label
export function Button(props){
    return(
        <label>
            {props.label}
            <input type="button" value={props.value} />
        </label>
    );
}
Button.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
}

export default Button;