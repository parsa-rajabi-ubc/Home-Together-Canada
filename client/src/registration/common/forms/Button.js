import React from 'react';
import PropTypes from "prop-types";


function Button(props){
    return(
        <label>
            {props.label}
            <input type="button" value={props.value} />
        </label>
    );
}
Button.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
}

export default Button