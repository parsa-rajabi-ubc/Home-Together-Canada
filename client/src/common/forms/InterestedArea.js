/**
 * @Author:     Parsa Rajabi
 * @Created:    2020.11.23
 *
 * @Description: Interested Are component for Member Profile
 *
 */

import React, {useState, useEffect} from 'react';
import radii from "./Radii";
import {getProvinces, getCities} from "../utils/locationUtils";
import Dropdown from "./Dropdown";
import PropTypes from "prop-types";
import Button from "./Button";
import {MdDeleteForever} from 'react-icons/md';
import {dropdownDefaultCSS} from "../../css/dropdownCSSUtil"


function InterestedArea(props) {
    const {onChange, givenAreasOfInterest, areasOfInterestError} = props;
    const [extraAreas, setExtraAreas] = useState(givenAreasOfInterest || [{province: "", city: "", radius: ""}]);

    const handleRemoveClick = index => {
        const list = [...extraAreas];
        list.splice(index, 1);
        setExtraAreas(list);
    };
    const handleAddClick = () => {
        const list = [...extraAreas];
        list.push({province: undefined, city: undefined, radius: undefined});
        setExtraAreas(list);
    };
    const handleAreaProvinceChange = (e, index) => {
        const {value} = e;
        const list = [...extraAreas];
        list[index].province = value;
        list[index].city = undefined;
        list[index].radius = undefined;
        setExtraAreas(list);
    };
    const handleAreaCityChange = (e, index) => {
        const {value} = e;
        const list = [...extraAreas];
        list[index].city = value;
        list[index].radius = undefined;
        setExtraAreas(list);
    };
    const handleAreaRadiusChange = (e, index) => {
        const {value} = e;
        const list = [...extraAreas];
        list[index].radius = value;
        setExtraAreas(list);
    };
    useEffect(() => {
        onChange(
            extraAreas
        )
    }, [extraAreas]);

    return (
        <div>
            <section className={`${areasOfInterestError && "border rounded-lg p-1 border-red-500"}`}>
                {extraAreas.map((currentAreaValues, index) => {
                    return (
                        <div key={index} className="grid grid-cols-9 gap-x-2">
                            <div className="col-start-1 col-end-4">
                                <Dropdown isSearchable={true} placeholder={"Province"}
                                          name="province"
                                          options={getProvinces()}
                                          onChange={e => handleAreaProvinceChange(e, index)}
                                          intialSelection={{label: extraAreas[index].province, value: extraAreas[index].province}}
                                          dropdownCSS={dropdownDefaultCSS}
                                />
                                <span>{currentAreaValues[index]}</span>
                            </div>
                            <div className="col-start-4 col-end-8">
                                {currentAreaValues.province &&
                                <Dropdown isSearchable={true} placeholder={"City"}
                                          name="city"
                                          options={getCities(currentAreaValues.province)}
                                          onChange={e => handleAreaCityChange(e, index)}
                                          intialSelection={{label: extraAreas[index].city, value: extraAreas[index].city}}
                                          dropdownCSS={dropdownDefaultCSS}
                                />}
                            </div>
                            <div className="col-start-8 col-end-10">
                                {currentAreaValues.city && <Dropdown isSearchable={true} placeholder={"Within Radius"}
                                                                     name="radius"
                                                                     options={radii}
                                                                     onChange={e => handleAreaRadiusChange(e, index)}
                                                                     intialSelection={{label: extraAreas[index].radius, value: extraAreas[index].radius}}
                                                                     dropdownCSS={dropdownDefaultCSS}
                                />}
                            </div>
                            <div className="col-start-10 col-end-auto">
                                {extraAreas.length !== 1 &&
                                <MdDeleteForever color="#DB4437" size="40" onClick={() => handleRemoveClick(index)}/>}
                            </div>
                            <div className="col-start-1 col-end-4">
                                {extraAreas.length - 1 === index &&
                                <Button className="btn btn-green text-sm py-2 " value="Add Another Location"
                                        onClick={() => {
                                            handleAddClick()
                                        }}/>}
                            </div>
                        </div>
                    );
                })}
            </section>
        </div>

    );
}

InterestedArea.propTypes = {
    onChange: PropTypes.func,
    givenAreasOfInterest: PropTypes.array,
    areasOfInterestError: PropTypes.bool,
}

export default InterestedArea;