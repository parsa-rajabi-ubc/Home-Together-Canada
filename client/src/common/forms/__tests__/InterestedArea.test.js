/**
 * @Author:     Rachelle Gelden
 * @Created:    2021.05.29
 *
 * @Description: InterestedArea unit tests
 *
 */

import React from 'react';
import renderer from  'react-test-renderer'
import InterestedArea from "../InterestedArea";
import Button from "../Button";

describe('InterestedArea component', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // given
            const onChange = jest.fn();
            const givenAreasOfInterest = [
                { province: 'AB', city: 'Calgary', radius: 50 },
                { province: 'AB', city: 'Drumheller', radius: 60 }
            ];
            const areasOfInterestError = undefined;
            const max = 5;

            // when
            const component = renderer.create(
                    <InterestedArea
                        onChange={onChange}
                        givenAreasOfInterest={givenAreasOfInterest}
                        areasOfInterestError={areasOfInterestError}
                        max={max}
                    />
                ).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
    });

    describe('Add Another Location button', () => {
        it('should show the Add Location button when there are less areas of interest than the max', () => {
            // given
            const onChange = jest.fn();
            const givenAreasOfInterest = [
                { province: 'AB', city: 'Calgary', radius: 50 },
                { province: 'AB', city: 'Drumheller', radius: 60 }
            ];
            const areasOfInterestError = undefined;
            const max = 5;

            // when
            const testInstance = renderer.create(
                <InterestedArea
                    onChange={onChange}
                    givenAreasOfInterest={givenAreasOfInterest}
                    areasOfInterestError={areasOfInterestError}
                    max={max}
                />
            ).root;
            const addLocationButtonText = testInstance.findByType(Button).props.value;

            // then
            expect(addLocationButtonText).toBe('Add Another Location');
        });
        it('should not show the Add Location button when the number of areas of interest is equal to the max', () => {
            // given
            const onChange = jest.fn();
            const givenAreasOfInterest = [
                { province: 'AB', city: 'Calgary', radius: 50 },
                { province: 'AB', city: 'Drumheller', radius: 60 },
                { province: 'AB', city: 'Edmonton', radius: 70 },
                { province: 'AB', city: 'Milk River', radius: 80 },
                { province: 'AB', city: 'Lethbridge', radius: 90 }
            ];
            const areasOfInterestError = undefined;
            const max = 5;

            // when
            const testInstance = renderer.create(
                <InterestedArea
                    onChange={onChange}
                    givenAreasOfInterest={givenAreasOfInterest}
                    areasOfInterestError={areasOfInterestError}
                    max={max}
                />
            ).root;
            const addLocationButtonText = testInstance.findAllByType(Button);

            // then
            expect(addLocationButtonText.length).toBe(0);
        });
    });
});
