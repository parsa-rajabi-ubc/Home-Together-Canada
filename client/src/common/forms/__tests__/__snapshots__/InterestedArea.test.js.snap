// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestedArea component Snapshot test should render correctly regardless of properties 1`] = `
<div>
  <section
    className="undefined"
  >
    <div
      className="grid grid-cols-9 gap-x-2"
    >
      <div
        className="col-start-1 col-end-4"
      >
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1jumw1e-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1uccc91-singleValue"
                >
                  AB
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="province"
              type="hidden"
              value="AB"
            />
          </div>
        </div>
        <span />
      </div>
      <div
        className="col-start-4 col-end-8"
      >
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1jumw1e-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1uccc91-singleValue"
                >
                  Calgary
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-3-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="city"
              type="hidden"
              value="Calgary"
            />
          </div>
        </div>
      </div>
      <div
        className="col-start-8 col-end-10"
      >
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1jumw1e-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1uccc91-singleValue"
                >
                  50km
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-4-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="radius"
              type="hidden"
              value="50km"
            />
          </div>
        </div>
      </div>
      <div
        className="col-start-10 col-end-auto"
      >
        <svg
          color="#DB4437"
          fill="currentColor"
          height="40"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#DB4437",
            }
          }
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </div>
      <div
        className="col-start-1 col-end-4"
      />
    </div>
    <div
      className="grid grid-cols-9 gap-x-2"
    >
      <div
        className="col-start-1 col-end-4"
      >
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1jumw1e-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1uccc91-singleValue"
                >
                  AB
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-5-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="province"
              type="hidden"
              value="AB"
            />
          </div>
        </div>
        <span />
      </div>
      <div
        className="col-start-4 col-end-8"
      >
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1jumw1e-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1uccc91-singleValue"
                >
                  Drumheller
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-6-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="city"
              type="hidden"
              value="Drumheller"
            />
          </div>
        </div>
      </div>
      <div
        className="col-start-8 col-end-10"
      >
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1jumw1e-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1uccc91-singleValue"
                >
                  60km
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-7-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="radius"
              type="hidden"
              value="60km"
            />
          </div>
        </div>
      </div>
      <div
        className="col-start-10 col-end-auto"
      >
        <svg
          color="#DB4437"
          fill="currentColor"
          height="40"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#DB4437",
            }
          }
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </div>
      <div
        className="col-start-1 col-end-4"
      >
        <label>
          <input
            className="btn btn-green text-sm py-2 "
            onClick={[Function]}
            type="button"
            value="Add Another Location"
          />
        </label>
      </div>
    </div>
  </section>
</div>
`;
